src/rtlib/fb_file.h

```diff
@@ -365,3 +365,68 @@ extern const UTF_8 __fb_utf8_bmarkTb[7];
 
 FBCALL int          fb_FileCopy         ( const char *source, const char *destination );
 FBCALL int          fb_CrtFileCopy      ( const char *source, const char *destination );
+
+
+
+#define FREAD_CHUNK_SIZE 1048576
+#define FWRITE_CHUNK_SIZE 1048576
+
+static __inline__ size_t FB_FREAD_LARGE( void *ptr, size_t nmemb, FILE *stream )
+{
+   size_t total = 0, nread;
+
+   /* if nmemb <1MB, use fread() directly */
+   if (nmemb < FREAD_CHUNK_SIZE) return fread( ptr, 1, nmemb, stream );
```

Comment:
These shortcut 'if's are pretty pointless, only making the functions longer.

Reply:
Yeah, you're probably right. When I originally wrote the function I kept separate size and nmemb parameters, and it was useful to bypass the more complex logic.
Now though, it just saves potentially a tiny amount of arithmetic.